src/components/ProductVersionLink/ProductVersionLink.tsx

```diff
@@ -11,8 +18,31 @@ interface IProductVersionLinkProps {
  *
  * @param productVersion - Product Version object
  */
-export const ProductVersionLink = ({ productVersion }: IProductVersionLinkProps) => (
-  <Link to={`/products/${productVersion.product?.id}/versions/${productVersion.id}`}>
-    {productVersion.product?.name} - {productVersion.version}
-  </Link>
-);
+export const ProductVersionLink = ({ productVersion }: IProductVersionLinkProps) => {
+  const serviceContainerProductVersion = useServiceContainer(productVersionApi.getProductVersion);
+  const serviceContainerProductVersionRunner = serviceContainerProductVersion.run;
```

Comment:
I understand that Build Configs provides only `ProductVersionRef` not containing the Product name and ID, but this component needs to remain dumb, therefore, the Version should be fetched one level up. The behavior is really weird (it accepts a Product Version and fetches the same Product Version) - even if one is a Ref and the other a normal Version, it is not right.

Reply:
If we want the product fetching process level-up to where we use this component, then the developer who uses the component will need to know whether the productVersion they passed to the component actually has a product object. This is not a good practice in terms of the encapsulation concept, so what I will do is to refactor the ProductVersionLink component to just accept the productVersionLink which will make it look better.

Reply:
That's not the best choice either, because the Version Link component should parse and create the link, not accept it (if that is what you meant).

A better solution is to ensure the component can accept only Version, where Product IS defined:
```tsx
interface ProductVersionWithProduct extends ProductVersion {
  product: Product;
}

interface IProductVersionLinkProps {
  productVersion: ProductVersionWithProduct;
}
```

or

```ts
interface IProductVersionLinkProps {
  productVersionId: string;
  productId: string;
}
```

Reply:
btw we right now use both approaches on our link components, some accept IDs and some entity objects, which is quite unfortunately inconsistent. @matedo1

Reply:
Hi both,

Still, I am not very convinced to accept redundant data that actually can be enough with one id. Our GET productVersion endpoint actually returns a ProductVersion object that contains also the Product object in it, which means taking only productVersion ID is enough, if we are accepting both id then what's the point of that? 

Thanks.
cc @patrikk0123 @matedo1

Reply:
Also just a reminder that we are not only parsing the URL to the product version but also displaying the product name and version name here, thus only the product id and productVersion id are not enough, we do need at least these names.

I do agree that a dumb component can be better, especially when maybe we are going to display a lot of product version links in a list. But given the situation that our backend sometimes can provide not a full version object but instead a ref version of the object (which is the root cause of why we had this discussion: there is no product information in the `ProductVersion` object in the `BuildConfig` entity) (see the long-history discussion of NCL-8124), seems at this moment we have no choice but fire a request to get these information we need(productVersion name, product id, product name).

If we have to send a request to the backend, I would suggest using my original implementation, which is  still accepting the productVersion data, but inside the component, if we find there is no product entity in the entity we then fire a request to get the ProductVersion entity. The good reason for doing this is that 1). other developers do not need to worry about whether this is a regular version of ProductVersion entity or ProductVersionRef that has no product entity, and 2).especially in the future we wish to totally remove the `ref` version of entity, and by that time it will be easy to just change the productVersion component to a dumb one without changing anywhere else.

I will revert this component back to the version I mentioned above first, and we can discuss further over that.

Reply:
You are right that the component also displays the title, so instead of passing IDs of Product and Version, I would pass actual entities Product and Version to the component. If the link is used with the Version entity, the Product in it can be passed, if it's VersionRef, the Product needs to be fetched in the component rendering the link.

So of course, if these required data (Product) are not available, they need to be fetched, but this component has to remain dumb so the fetch will be done one level up.

Also, the component might defaultly get the Procuct from the passed Version, like this
```ts
const ProductVersionLink = ({
  productVersion,
  product = productVersion.product
}) => ...
```